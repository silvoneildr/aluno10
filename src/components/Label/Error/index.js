import React from 'react';
import Label from '..';

const LabelError = React.memo(props => <Label color="error" {...props} />);

export default LabelError;
