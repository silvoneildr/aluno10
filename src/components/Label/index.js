import React from 'react';
import propTypes from 'prop-types';
import Span from './styles';

const Label = React.memo(({ children, ...rest }) => <Span {...rest}>{children}</Span>);

Label.propTypes = {
  children: propTypes.string.isRequired
};

export default Label;
