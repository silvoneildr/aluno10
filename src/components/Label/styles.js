import styled from 'styled-components';

const Span = styled.span`
  color: ${props => props.theme[props.color] || props.theme.primary};
  margin: 5px 0;
`;

export default Span;
