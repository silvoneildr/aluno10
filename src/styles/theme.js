export default {
  strongColor: '#052c4b',
  primaryColor: '#4548fd',
  secondaryColor: '#ffffff',
  lightColor: '#dcdee091',
  successColor: '#37da6b',
  warningColor: '#f0d74d',
  errorColor: '#f8562d'
};
