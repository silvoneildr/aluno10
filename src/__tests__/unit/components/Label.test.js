import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Label from 'Components/Label';
import LabelError from 'Components/Label/Error';
import theme from 'Root/styles/theme';

afterEach(cleanup);

describe('<Label />', () => {
  it('should render correclty', () => {
    const { container } = render(<Label theme={theme}>Informação</Label>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('<LabelError />', () => {
  it('should render correclty', () => {
    const { container } = render(<LabelError theme={theme}>Informação</LabelError>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
