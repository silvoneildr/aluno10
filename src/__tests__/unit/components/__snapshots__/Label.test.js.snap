// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render correclty 1`] = `
.c0 {
  color: #4548fd;
  margin: 5px 0;
}

<span
  class="c0"
>
  Informação
</span>
`;

exports[`<LabelError /> should render correclty 1`] = `
.c0 {
  color: #FF9494;
  margin: 5px 0;
}

<span
  class="c0"
  color="error"
>
  Informação
</span>
`;
